### PR TITLE
Extract Currency.Amount.t into a separate, compatible library

### PR DESCRIPTION
**Pull request by Firobe**
PoC for #11101

### CODE REVIEW

File: src/lib/currency/currency.ml [hunk 1]
```diff
@@ -595,62 +596,146 @@ module Fee = struct
 end
 
 module Amount = struct
-  module T =
-    Make
-      (Unsigned_extended.UInt64)
-      (struct
-        let length = currency_length
-      end)
+  (* The basic type t of this module is defined in Mina_wire_types but is
```
Comment (mrmr1993) [inline src/lib/currency/currency.ml:599]: This comment will end up being repeated everywhere if we want to be thorough. Would it make more sense to move it into a top-level doc-comment in `Mina_wire_types`, and have a cross-reference to there each time?

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/currency/currency.ml (hunk 1 above)
Comment (Firobe) [reply src/lib/currency/currency.ml:599]: Yeah, I think I'll write a simple `mina_wire_types.mld` file with high-level documentation for the whole library.

**Comment by Firobe** [comment]
Added an `optcomp` branching to ensure compatibility with `src/nonconsensus` as well.

**Review by mrmr1993 — APPROVED** [review]

**Comment by Firobe** [comment]
Added some short [high-level documentation](https://github.com/MinaProtocol/mina/blob/tweag/mina-wire-types/src/lib/mina_wire_types/index.mld).

**Pull request merged**